backend/src/agents/web_agent.py

```diff
@@ -0,0 +1,65 @@
+import logging
+from src.prompts import PromptEngine
+from .agent_types import Parameter
+from .agent import Agent, agent
+from .tool import tool
+from src.utils import Config
+from src.utils import update_scratchpad
+from src.utils.web_utils import search_urls, scrape_content, summarise_content
+from .validator_agent import ValidatorAgent
+logger = logging.getLogger(__name__)
+config = Config()
+
+engine = PromptEngine()
+
+
+@tool(
+    name="web_general_search",
+    description="Search the internet based on the query provided and then get the meaningful answer from the content found",
+    parameters={
+        "search_query": Parameter(
+            type="string",
+            description="The search query to find information on the internet",
+        ),
+    },
+)
+def web_general_search(search_query, llm, model) -> list:
+
+    urls = []
+    urls = search_urls(search_query)
+    logger.info(f"URLs found: {urls}")
+    if not urls:
+        return ["No relevant information found on the internet for the given query."]
+    update_scratchpad(result=urls)
```

Comment:
do we want to include additional info like agent_name when updating the scratchpad?

Reply:
Makes sense, have included it now.